src/main/java/core/basesyntax/strategy/impl/ReturnOperation.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.Operation;
+import core.basesyntax.strategy.OperationException;
+
+public class ReturnOperation implements Operation {
+    private final StorageDao storageDao;
+
+    public ReturnOperation(StorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public void proceed(FruitTransaction fruitTransaction) {
+        if (storageDao.get(fruitTransaction.getFruitName()) == null) {
+            throw new OperationException("Fruit " + fruitTransaction.getFruitName()
+                    + " doesn't exist");
```

Comment:
This check is redundant because in the case of a return operation, if the fruit does not exist in the storage, it should be added with the returned quantity. The current exception will prevent the return operation from being processed correctly.